controllers/articles.controllers.js

```diff
@@ -3,12 +3,22 @@ const {
     selectArticleById,
     selectCommentsByArticleId,
     insertCommentByArticleId,
-    selectUserByUsername,
+    checkTopicExists,
 } = require("../models/articles.models")
 
-exports.getAllArticles = (_, res, next) => {
-    selectAllArticles()
-        .then((articles) => {
+exports.getAllArticles = (req, res, next) => {
+    const { topic } = req.query
+    const { sort_by } = req.query
+    const { order } = req.query
+    const promises = [selectAllArticles(topic, sort_by, order)]
+
+    if (topic) {
+        promises.push(checkTopicExists(topic))
+    }
+    Promise.all(promises)
```

Comment:
I really like the promise.all here this is a great way of handling the check whether a promise exists